backend.go

```diff
@@ -5,45 +5,78 @@ package fakemachine
 
 import (
 	"fmt"
+	"sort"
 )
 
-// A list of backends which are implemented
+// Main list of backends vs. their order in the "auto" algoritmn.
+func implementedBackends(m *Machine) map[int]backend {
+	return map[int]backend{
+		10: newKvmBackend(m),
+		20: newUmlBackend(m),
+		30: newQemuBackend(m),
+	}
+}
+
+/* maps are iterated randomly in Go - calculate and store the sorted backend
+ * order in a seperate array
```

Comment:
Nit: s/seperate/separate/
Another go maps trivia - they cannot be const :confused: 

In case it matters: I like the algorithmic approach but I wonder if it's worth it. Are we having for (m)any new backends in the future?

Reply:
Fixed the spelling mistake.
I added this approach due to when the qemu backend was added it was missed out from the BackendNames() function - there are no real plans to add other backends but I still think this is useful to generally improve the code as the priorities may also change in future?

.... other than my idea/hack implementation of a virtualbox backend i don't think there are many other options? ;-)